component/otelcol/receiver/vcenter/vcenter.go

```diff
@@ -0,0 +1,345 @@
+// Package vcenter provides an otelcol.receiver.vcenter component.
+package vcenter
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/component/otelcol/receiver"
+	otel_service "github.com/grafana/agent/service/otel"
+	"github.com/grafana/river/rivertypes"
+	"github.com/mitchellh/mapstructure"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/vcenterreceiver"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configopaque"
+	otelextension "go.opentelemetry.io/collector/extension"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:          "otelcol.receiver.vcenter",
+		Args:          Arguments{},
+		NeedsServices: []string{otel_service.ServiceName},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := vcenterreceiver.NewFactory()
+			return receiver.New(opts, fact, args.(Arguments))
+		},
+	})
+}
+
+type MetricConfig struct {
+	Enabled bool `river:"enabled,attr"`
+}
+
+func (r *MetricConfig) Convert() map[string]interface{} {
+	if r == nil {
+		return nil
+	}
+
+	return map[string]interface{}{
+		"enabled": r.Enabled,
+	}
+}
+
+// SetToDefault implements river.Defaulter.
+func (args *MetricConfig) SetToDefault() {
+	*args = MetricConfig{
+		Enabled: true,
+	}
+}
```

Comment:
I think we can remove this because "enabled" is mandatory. Hence, there is no need for a default value.